File: python_ToSXs/python/06-thaphinh/95.py
height = 7

shape = ""
record = ""

for i in range(1, height + 1):
    character = "#" if (i % 2 == 0) else "o"
    around = " " * (height - i)
    center = (str(i) + character) * i

    center = center[0:len(center) - 1]
    record = around + center + around

    shape += record + "\n"

for i in range(1, int(height / 2) + 1):
    space1 = " " * int((height + 1) / 2)
    space2 = " " * (height - 4)
    record = space1 + "|" + space2 + "|"
    shape += record + "\n"

shape += "-" * (2 * height - 1)

print(shape)

File: cpp_e9zMZ/cpp/06-thaphinh/95.cpp
#include <iostream>

#include "zvn.h"
using namespace std;

int main() {
    int height = 7;
    string shape = "";
    string record = "";

    for (int i = 1; i <= height; i++) {
        string character = (i % 2 == 0) ? "#" : "o";
        string around = myRepeat(" ", height - i);
        string center = myRepeat(to_string(i) + character, i);

        center = center.substr(0, center.length() - 1);
        record = around + center + around;

        shape += record + "\n";
    }

    for (int i = 1; i <= height / 2; i++) {
        string space1 = myRepeat(" ", (height + 1) / 2);
        string space2 = myRepeat(" ", (height - 4));
        record = space1 + "|" + space2 + "|";
        shape += record + "\n";
    }

    shape += myRepeat("-", 2 * height - 1);
    cout << shape << endl;
    return 0;
}